### PR TITLE
Cache vendor in CI

### PR DESCRIPTION
**Pull request by clebs**
Confirm these statements before you submit your pull request:

- [x] I have read and submitted the required [Contributor Licence Agreements](https://github.com/kyma-project/community/blob/master/CONTRIBUTING.md#agreements-and-licenses).
- [x] This pull request follows the contributing guidelines, recommended Git workflow, and templates.
- [x] I have tested my changes.
- [x] I have updated the relevant documentation.
---

**Description**

Changes proposed in this pull request:

- Add caching of vendor folders when building golang components

### CODE REVIEW

**Comment by Abd4llA** [comment]
@clebs When will this cache gets invalidated ? When the step produces different files in hash?
how does it determine that vendor comes from hash but e.g. bin/<new bin> doesn't?

**Comment by clebs** [comment]
@Abd4llA this is a PoC for now, so I do not have full results yet.
Nevertheless, this cache is not invalidated, it gets updated on every build when `dep ensure` updates the contents of the `vendor` folder.

At the start of  the dependencies stage it is loaded, gets updated and is stored again at the end of the stage.

>how does it determine that vendor comes from hash but e.g. bin/ doesn't?
I am afraid I did not understand the question, could you rephrase it?

**Comment by clebs** [comment]
Closing PR, the cache plugin is not viable for a multiple parallel job scenario.
Parallel jobs write to the same cache causing race conditions, I/O errors and overwrites. No way found to have dedicated cache for each job.
